### PR TITLE
8 packages from savonet/ocaml-ffmpeg at 1.1.5

### PR DESCRIPTION
**Pull request by toots**
This pull-request concerns:
-`ffmpeg.1.1.5`: Bindings for the ffmpeg libraries
-`ffmpeg-av.1.1.5`: Bindings for the ffmpeg libraries -- top-level helpers
-`ffmpeg-avcodec.1.1.5`: Bindings for the ffmpeg avcodec library
-`ffmpeg-avdevice.1.1.5`: Bindings for the ffmpeg avdevice library
-`ffmpeg-avfilter.1.1.5`: Bindings for the ffmpeg avfilter library
-`ffmpeg-avutil.1.1.5`: Bindings for the ffmpeg avutil libraries
-`ffmpeg-swresample.1.1.5`: Bindings for the ffmpeg swresample library
-`ffmpeg-swscale.1.1.5`: Bindings for the ffmpeg swscale library

---
* Homepage: https://github.com/savonet/ocaml-ffmpeg
* Source repo: git+https://github.com/savonet/ocaml-ffmpeg.git
* Bug tracker: https://github.com/savonet/ocaml-ffmpeg/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

**Comment by mseri** [comment]
There is a pervasive failure:
```
# File "test/dune", line 5, characters 0-2203:
#  5 | (rule
#  6 |  (alias runtest)
#  7 |  (package ffmpeg)
# ....
# 58 |    (run %{read_metadata} A4.mp2 mp2 A4.ogg libvorbis)
# 59 |    (run %{interrupt})
# 60 |    (run %{test}))))
# (cd _build/default/test && ../examples/decode_audio.exe A4.flac flac A4.mp3 libmp3lame)
# Fatal error: exception Avutil.Error(Could not find C value for AV_CH_LAYOUT_T_TAB. Do you need to recompile the ffmpeg binding?)
# Raised at Avutil.(fun) in file "avutil/avutil.ml", line 85, characters 51-77
# Called from Avcodec.Audio.get_supported_channel_layouts in file "avcodec/avcodec.ml", line 274, characters 13-50
# Called from Avcodec.Audio.get_supported_channel_layouts in file "avcodec/avcodec.ml", line 274, characters 13-50
# Called from Avcodec.Audio.find_best_channel_layout in file "avcodec/avcodec.ml", line 278, characters 28-63
# Called from Dune__exe__Decode_audio in file "examples/decode_audio.ml", line 37, characters 23-75
```
Is it to be expected?

**Comment by toots** [comment]
> Reference

This looks like a regression in the upstream library. The channel layout API has been rewritten there. I'll address it by implementing the new API in a future version.

**Comment by toots** [comment]
> Do we need to add some specific upper bounds or conflicts to prevent this issue?

I'm not sure that depext supports version constraints for the upstream library at the moment, does it?

**Comment by kit-ty-kate** [comment]
no it doesn't

**Comment by kit-ty-kate** [comment]
Is this a related failure?
```
# File "test/dune", line 5, characters 0-2203:
#  5 | (rule
#  6 |  (alias runtest)
#  7 |  (package ffmpeg)
# ....
# 58 |    (run %{read_metadata} A4.mp2 mp2 A4.ogg libvorbis)
# 59 |    (run %{interrupt})
# 60 |    (run %{test}))))
# (cd _build/default/test && ../examples/encode_stream.exe S4.mp2 mp2)
# Fatal error: exception Avutil.Error(Invalid argument)
# Raised by primitive operation at Av.new_audio_stream in file "av/av.ml", line 297, characters 4-106
# Called from Dune__exe__Encode_stream in file "examples/encode_stream.ml", line 54, characters 4-134
```
We can see that on archlinux for example, which i would assume would have the most up-to-date version

**Comment by toots** [comment]
> Is this a related failure?
> 
> ```
> # File "test/dune", line 5, characters 0-2203:
> #  5 | (rule
> #  6 |  (alias runtest)
> #  7 |  (package ffmpeg)
> # ....
> # 58 |    (run %{read_metadata} A4.mp2 mp2 A4.ogg libvorbis)
> # 59 |    (run %{interrupt})
> # 60 |    (run %{test}))))
> # (cd _build/default/test && ../examples/encode_stream.exe S4.mp2 mp2)
> # Fatal error: exception Avutil.Error(Invalid argument)
> # Raised by primitive operation at Av.new_audio_stream in file "av/av.ml", line 297, characters 4-106
> # Called from Dune__exe__Encode_stream in file "examples/encode_stream.ml", line 54, characters 4-134
> ```
> 
> We can see that on archlinux for example, which i would assume would have the most up-to-date version

Yes I believe that this is the issue but I will have to investigate. I will also look at failing the whole build based on versions detected there, this should at least give a good user-friendly error report.

**Comment by toots** [comment]
Hi! Any update on this? We bumped the dependency version on `liquidsoap` to force installing this one with the `2.1.1` release so, without it, this prevents a whole lot of functionalities.

I plan on addressing the issues with the upstream library version in the next release.

Thanks for y'all help!

**Comment by kit-ty-kate** [comment]
> I plan on addressing the issues with the upstream library version in the next release.

If so, should we disable the tests? (as they seem to fail unconditionally at the moment)

**Comment by toots** [comment]
They do pass in a fairly good amount of cases don't they?

**Comment by mseri** [comment]
Not `ffmpeg.1.1.5`, that fails everywhere, all the rest yes. We are suggesting just to disable tests in `ffmpeg.1.1.5`

File: packages/ffmpeg/ffmpeg.1.1.5/opam [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
```
Comment (mseri) [inline packages/ffmpeg/ffmpeg.1.1.5/opam:31]:
```suggestion
```

Comment (mseri) [reply packages/ffmpeg/ffmpeg.1.1.5/opam:31]: The commit was not disabling the tests for ffmpeg which is the one that fails.

File: packages/ffmpeg-av/ffmpeg-av.1.1.5/opam [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@doc" {with-doc}
```
Comment (mseri) [inline packages/ffmpeg-av/ffmpeg-av.1.1.5/opam:32]:
```suggestion
    "@runtest" {with-test}
    "@doc" {with-doc}
```

**Comment by toots** [comment]
Thanks!

**Pull request merged**